src/registry_testing.jl

```diff
@@ -238,6 +238,32 @@ function test(path=pwd(); registry_deps::Vector{<:AbstractString}=String[])
                 else
                     @debug "Compat.toml file does not exist" compatfile
                 end
+
+                # https://github.com/JuliaRegistries/RegistryCI.jl/issues/522
+                # For every version, each compat entry has a corresponding deps entry.
+                #
+                # Note: it is legal for a package to have Compat.toml but not have
+                # Deps.toml. Specifically, this is legal if (and only if) the package
+                # does not have any dependencies, and does have a compat entry for `julia`.
```

Comment:
Shouldn't WeakDeps.toml be part of this reasoning too?

Reply:
Yeah, we should do the same thing for WeakDeps/WeakCompat.

Reply:
Done. We now do this for both Deps/Compat and also for WeakDeps/WeakCompat.

For Deps/Compat, it is legal for Compat.toml to exist but Deps.toml to not exist - specifically, if a package has no dependencies, and it has a compat entry for `julia`, then it will have Compat.toml but not Deps.toml.

However, if I understand correctly, it is never legal for a package to have Compat.toml but not have Deps.toml.

Reply:
I think that this has been resolved by https://github.com/JuliaRegistries/RegistryCI.jl/pull/525/commits/631ed08338e912a4e921e33a4bfbd03d942c9a1e